tensorrt_llm/_torch/auto_deploy/utils/multi_stream_utils.py

```
@@ -0,0 +1,242 @@
+# Previously, CudaStreamManager and the custom ops that use the cuda streams and events were
+# placed in custom_ops folder. However doing so resulted in CudaStreamManager
+# being created only in the parent process, but we need each rank to have its own CudaStreamManager that
+# manages the cuda streams and events for that rank. Placing the logic to instantiate
+# CudaStreamManager and the custom ops that use the cuda streams and events at the transform level ensures that
+# each rank has its own CudaStreamManager since each rank applies the transform independently.
+
+
+class CudaStreamManager(metaclass=_Singleton):
+    AUX_STREAM_NAME = "aux"
+    MAIN_STREAM_NAME = "main"
+    devices: List[torch.device] = []
```

Ruff (0.15.1)

[warning] Mutable default value for class attribute

(RUF012)